src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicConfiguration.cs

```diff
@@ -117,16 +117,43 @@ private static unsafe MsQuicSafeHandle Create(QuicConnectionOptions options, QUI
 #pragma warning restore SYSLIB0040
 
         QUIC_SETTINGS settings = default(QUIC_SETTINGS);
+
         settings.IsSet.PeerUnidiStreamCount = 1;
         settings.PeerUnidiStreamCount = (ushort)options.MaxInboundUnidirectionalStreams;
+
         settings.IsSet.PeerBidiStreamCount = 1;
         settings.PeerBidiStreamCount = (ushort)options.MaxInboundBidirectionalStreams;
+
         if (options.IdleTimeout != TimeSpan.Zero)
         {
             settings.IsSet.IdleTimeoutMs = 1;
-            settings.IdleTimeoutMs = options.IdleTimeout != Timeout.InfiniteTimeSpan ? (ulong)options.IdleTimeout.TotalMilliseconds : 0;
+            settings.IdleTimeoutMs = options.IdleTimeout != Timeout.InfiniteTimeSpan
+                ? (ulong)options.IdleTimeout.TotalMilliseconds
+                : 0; // 0 disables the timeout
         }
 
+        settings.IsSet.KeepAliveIntervalMs = 1;
+        settings.KeepAliveIntervalMs = options.KeepAliveInterval != Timeout.InfiniteTimeSpan
+            ? (uint)options.KeepAliveInterval.TotalMilliseconds
+            : 0; // 0 disables the keepalive
+
+        settings.IsSet.ConnFlowControlWindow = 1;
+        settings.ConnFlowControlWindow = (uint)options.InitialReceiveWindowSizes.Connection;
+
+        settings.IsSet.StreamRecvWindowBidiLocalDefault = 1;
```

Comment:
Correct, the MsQuic version where this will be released should be 2.4. I was thinking whether we need to raise the minimum version of the library. AFAIK, it should work with older versions, the settings would just be ignored.